test/AzExtFsExtra.test.ts

```diff
@@ -0,0 +1,139 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import assert = require('assert');
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import { Uri, workspace, WorkspaceFolder } from 'vscode';
+import { AzExtFsExtra, getRandomHexString } from "../extension.bundle";
+import { assertThrowsAsync } from './assertThrowsAsync';
+
+
+suite('AzExtFsExtra', function (this: Mocha.Suite): void {
+    let workspacePath: string;
+    let testFolderPath: string;
+    let workspaceFilePath: string;
+
+    const indexHtml: string = 'index.html';
+
+    const nonExistingPath: string = ' ./path/does/not/exist';
+    const nonExistingFilePath = path.join(nonExistingPath, indexHtml);
+
+    suiteSetup(function (this: Mocha.Context): void {
+        testFolderPath = `azExtFsExtra${getRandomHexString()}`
+        const workspaceFolders: readonly WorkspaceFolder[] | undefined = workspace.workspaceFolders;
+        if (!workspaceFolders) {
+            throw new Error("No workspace is open");
+        }
+
+        workspacePath = workspaceFolders[0].uri.fsPath;
+        workspaceFilePath = path.join(workspacePath, indexHtml);
+    });
+
+    test('pathExists for directory', async () => {
+        assert.strictEqual(await AzExtFsExtra.pathExists(workspacePath), fs.existsSync(workspacePath));
+        assert.strictEqual(await AzExtFsExtra.pathExists(nonExistingPath), fs.existsSync(nonExistingPath));
+    });
+
+    test('pathExists for file', async () => {
+        assert.strictEqual(await AzExtFsExtra.pathExists(workspaceFilePath), fs.existsSync(workspaceFilePath));
+        assert.strictEqual(await AzExtFsExtra.pathExists(nonExistingFilePath), fs.existsSync(nonExistingFilePath));
+    });
+
+    test('isDirectory properly detects folders', async () => {
+        assert.strictEqual(await AzExtFsExtra.isDirectory(workspacePath), isDirectoryFs(workspacePath));
+        assert.strictEqual(await AzExtFsExtra.isDirectory(workspaceFilePath), isDirectoryFs(workspaceFilePath));
```

Comment:
`isDirectory` is what this test is for, so we can use that from `AzExtFsExtra`. Since we know what the "expected" value should be, we can hard-code that to `true` and `false`
```suggestion
        assert.strictEqual(await AzExtFsExtra.isDirectory(workspacePath), true);
        assert.strictEqual(await AzExtFsExtra.isDirectory(workspaceFilePath), false);
```